components/experimental/src/dimension/percent/format.rs

```diff
@@ -0,0 +1,218 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use fixed_decimal::{FixedDecimal, Sign};
+
+use crate::alloc::borrow::ToOwned;
+use alloc::borrow::Cow;
+use writeable::Writeable;
+
+use crate::dimension::provider::percent::PercentEssentialsV1;
+
+use super::options::{Display, PercentFormatterOptions};
+
+struct Append<W1, W2>(W1, W2);
+// This allows us to combines two writeables together.
+impl<W1: Writeable, W2: Writeable> Writeable for Append<W1, W2> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        self.0.write_to(sink)?;
+        self.1.write_to(sink)
+    }
+}
+
+pub struct FormattedPercent<'l> {
+    pub(crate) value: &'l FixedDecimal,
+    pub(crate) essential: &'l PercentEssentialsV1<'l>,
+    pub(crate) options: &'l PercentFormatterOptions,
+}
+
+impl<'l> Writeable for FormattedPercent<'l> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        // Removing the sign from the value
+        let abs_value = match self.value.sign() {
+            Sign::Negative => self.value.clone().with_sign(Sign::None),
+            _ => self.value.to_owned(),
+        };
+
+        match self.options.display {
+            // In the Standard display, we take the positive pattern only when the value is positive.
+            Display::Standard => {
+                if self.value.sign() == Sign::Negative {
+                    self.essential
+                        .negative_pattern
+                        .interpolate((abs_value, &self.essential.minus_sign))
```

Comment:
Issue, here and elsewhere: instead of using `impl Writeable for FixedDecimal`, you should be using `impl Writeable for FormattedFixedDecimal`, which you can obtain from a FixedDecimalFormatter, which you should include as a field of the PercentFormatter.

Without the FixedDecimalFormatter, you won't get localized digits, decimal separators, or grouping separators.

You can test this with locales such as `bn` and `ar-EG`.